src/main/java/ru/spbstu/service/UserService.java

```diff
@@ -0,0 +1,60 @@
+package ru.spbstu.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import ru.spbstu.model.Role;
+import ru.spbstu.model.User;
+import ru.spbstu.repository.UserRepository;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserService implements UserDetailsService {
+    @Autowired
+    UserRepository userRepository;
+    @Autowired
+    BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userRepository.findByUsername(username);
+        if (user == null) {
+            throw new UsernameNotFoundException("User not found");
+        }
+        return user;
+    }
+
+    public User findUserById(Long userId) {
+        Optional<User> userFromDb = userRepository.findById(userId);
```

Comment:
Why not return `Optional<User>` instead of creating the user using the constructor without parameters?

Reply:
this method should throw the `notFound` exception if a user wasn't found IMHO